subsys/bluetooth/audio/ascs.c

```diff
@@ -1490,41 +1501,138 @@ static bool ascs_codec_store_metadata(struct bt_data *data, void *user_data)
 	struct bt_codec *codec = user_data;
 	struct bt_codec_data *meta;
 
-	if (codec->meta_count >= ARRAY_SIZE(codec->meta)) {
-		BT_ERR("No slot available for Codec Config Metadata");
-		return false;
-	}
-
 	meta = &codec->meta[codec->meta_count];
-
-	if (data->data_len > sizeof(meta->value)) {
-		BT_ERR("Not enough space for Codec Config Metadata: %u > %zu",
-		       data->data_len, sizeof(meta->value));
-		return false;
-	}
-
-	BT_DBG("#%u type 0x%02x len %u", codec->meta_count, data->type,
-	       data->data_len);
-
 	meta->data.type = data->type;
 	meta->data.data_len = data->data_len;
 
 	/* Deep copy data contents */
 	meta->data.data = meta->value;
 	(void)memcpy(meta->value, data->data, data->data_len);
 
-	BT_HEXDUMP_DBG(meta->value, data->data_len, "data");
+	BT_DBG("#%zu: data: %s",
+	       codec->meta_count,
+	       bt_hex(meta->value, data->data_len));
 
 	codec->meta_count++;
 
 	return true;
 }
 
+struct ascs_parse_result {
+	int err;
+	size_t count;
+	const struct bt_audio_ep *ep;
+};
+
+static bool ascs_parse_metadata(struct bt_data *data, void *user_data)
+{
+	struct ascs_parse_result *result = user_data;
+	const struct bt_audio_ep *ep = result->ep;
+	const uint8_t data_len = data->data_len;
+	const uint8_t data_type = data->type;
+	const uint8_t *data_value = data->data;
+
+	result->count++;
+
+	BT_DBG("#%u type 0x%02x len %u", result->count, data_type, data_len);
+
+	if (result->count > CONFIG_BT_CODEC_MAX_METADATA_COUNT) {
```

Comment:
I am certain this will cause problems just like *unsupported* metadata type. Does the spec mandates the maximum number of metadata fields? Why not to parse the metadata anyway to get the *known* types?

Reply:
> Why not to parse known specific metadata types only?

You mean truncate the metadata to fit the memory allocated in Zephyr?

I guess we can do that, but in the case of e.g. CAP (which mandates e.g. the streaming context metadata), then we may end up truncating required metadata. 

Perhaps ideally the stack wouldn't need to store this data, and probably just provide the metadata as a `bt_data` to the upper layers as is, which I guess is similar to what is being requested in https://github.com/zephyrproject-rtos/zephyr/issues/42860

Reply:
So why not to parse and store required metadata only?

Reply:
> So why not to parse and store required metadata only?

We could/should do that, but that seems outside the scope of this PR, as that is changing how we store metadata (and then we need to refactor all metadata handling in ASCS). 

I suggest to postpone this to https://github.com/zephyrproject-rtos/zephyr/issues/42860